driver-core/src/main/java/com/datastax/driver/core/tracing/NoopTracingInfoFactory.java

```diff
@@ -0,0 +1,39 @@
+package com.datastax.driver.core.tracing;
+
+import com.datastax.driver.core.ConsistencyLevel;
+
+public class NoopTracingInfoFactory implements TracingInfoFactory {
+
+  private static class NoopTracingInfo implements TracingInfo {
+    @Override
+    public void setNameAndStartTime(String name) {}
+
+    @Override
+    public void setConsistencyLevel(ConsistencyLevel consistency) {}
+
+    @Override
+    public void setStatementType(String statementType) {}
+
+    @Override
+    public void recordException(Exception exception) {}
+
+    @Override
+    public void setStatus(StatusCode code, String description) {}
+
+    @Override
+    public void setStatus(StatusCode code) {}
+
+    @Override
+    public void tracingFinished() {}
+  }
+
+  @Override
+  public TracingInfo buildTracingInfo() {
+    return new NoopTracingInfo();
```

Comment:
Small nit: instead of creating a new `NoopTracingInfo` object every time, you could use a Singleton pattern:

```java
    private static NoopTracingInfo INSTANCE = new NoopTracingInfo();

    public TracingInfo buildTracingInfo() {
        return INSTANCE;
    }
```

Reply:
And `final` would fit here as well, I guess.